### DISCUSSION TITLE
How to debug MSW in my React tests?

### DISCUSSION THREAD

**Discussion by jrock2004** [mswjs/msw#1385] [Q&A] [answered]
I have a component that will show loading spinner till the api return. My test is waiting for that to be replaced with content from the API but it never happens. How can I debug in test mode to even see if MSW is working and or what could be going wrong with the API call

**Comment by kettanaito** [comment, marked as answer]
Hey, @jrock2004.

We have a [documentation recipe](https://mswjs.io/docs/recipes/debugging-uncaught-requests) on debugging uncaught requests. You can use it also to debug if MSW is working at all by setting the `onUnhandledRequest: "error"` option in `server.listen()` and making some unhandled arbitrary request. If it errors, MSW works; if it doesn't, your setup is incorrect and you should revisit [Node integration instructions](https://mswjs.io/docs/getting-started/integrate/node).
